### PR TITLE
FIX-1013: compress as algo 2

### PR DESCRIPTION
**Pull request by DenisYaroshevskiy**
The only thing that's left customising `compress_store` is zip.
That breaks in too many places.

Introduced `store_equivalent`.
It's better like this over purely customising `store` - because we want, for example, to convert the value once.

store customisation is left in `zip`, `reverse` and `backward` views.
In zip it will stay for sure. reverse and backward - maybe smth will be done.

Also https://github.com/jfalcou/eve/issues/1037 should be fixed with this PR.

### CODE REVIEW

File: include/eve/detail/compress/simd/common/compress_using_masks.hpp [hunk 1]
```diff
@@ -118,7 +118,7 @@ namespace eve::detail
   {
     if constexpr ( kumi::product_type<T> )
     {
-      return wide<T, N>{ kumi::map(compress_using_masks_shuffle, v, num) };
+      return wide<T, N>{ kumi::map([num](auto v_) { return compress_using_masks_shuffle(v_, num); }, v) };
```
Comment (jfalcou) [inline include/eve/detail/compress/simd/common/compress_using_masks.hpp:121]: shouldn't it be a function object for inlining purpose ?

File: include/eve/traits/value_type.hpp [hunk 2]
```diff
@@ -67,5 +67,6 @@ namespace eve
   //================================================================================================
 
   template <typename T>
+    requires (!value<T>)
```
Comment (jfalcou) [inline include/eve/traits/value_type.hpp:70]: why this change ?

File: include/eve/detail/compress/simd/common/compress_using_masks.hpp (hunk 1 above)
Comment (DenisYaroshevskiy) [reply include/eve/detail/compress/simd/common/compress_using_masks.hpp:121]: Yeah it should. Let me do that in the next pr

File: include/eve/traits/value_type.hpp (hunk 2 above)
Comment (DenisYaroshevskiy) [reply include/eve/traits/value_type.hpp:70]: It tries to get the value of doible and that results in a hard error, not sfinae error.

I don't have a concept that'd distinguish value and pointer here. So I did !value

Comment (jfalcou) [reply include/eve/traits/value_type.hpp:70]: OK

File: include/eve/detail/compress/simd/common/compress_using_masks.hpp (hunk 1 above)
Comment (jfalcou) [reply include/eve/detail/compress/simd/common/compress_using_masks.hpp:121]: Perfect

**Review by jfalcou — APPROVED** [review]
